quipucords/tests/api/scantask/test_scantask.py

```diff
@@ -47,202 +48,195 @@ def scan_job(scan: Scan) -> ScanJob:
-            "systems_scanned": 0,
-            "systems_failed": 0,
-            "systems_unreachable": 0,
-        }
-        assert expected == json_task
-
-    def test_successful_start(self, scan_job, source):
-        """Create a scan task and start it."""
-        task = ScanTask.objects.create(
-            job=scan_job,
-            source=source,
-            scan_type=ScanTask.SCAN_TYPE_CONNECT,
-            status=ScanTask.PENDING,
-        )
-        start_time = datetime.utcnow()
-        task.status_start()
-        assert messages.ST_STATUS_MSG_RUNNING == task.status_message
-        assert task.status == ScanTask.RUNNING
-        assert start_time.replace(microsecond=0) == task.start_time.replace(
-            microsecond=0
-        )
+def test_successful_create(scan_job, source):
+    """Create a scan task and serialize it."""
+    task = ScanTask.objects.create(
+        job=scan_job,
+        source=source,
+        scan_type=ScanTask.SCAN_TYPE_CONNECT,
+        status=ScanTask.PENDING,
+    )
+    serializer = ScanTaskSerializer(task)
+    json_task = serializer.data
+    expected = {
+        "sequence_number": 0,
+        "source": source.id,
+        "scan_type": ScanTask.SCAN_TYPE_CONNECT,
+        "status": "pending",
+        "status_message": messages.ST_STATUS_MSG_PENDING,
+        "systems_count": 0,
+        "systems_scanned": 0,
+        "systems_failed": 0,
+        "systems_unreachable": 0,
+    }
+    assert expected == json_task
 
-    def test_successful_restart(self, scan_job, source):
-        """Create a scan task and restart it."""
-        task = ScanTask.objects.create(
-            job=scan_job,
-            source=source,
-            scan_type=ScanTask.SCAN_TYPE_CONNECT,
-            status=ScanTask.PENDING,
-        )
-        task.status_restart()
-        assert messages.ST_STATUS_MSG_RESTARTED == task.status_message
-        assert task.status == ScanTask.PENDING
 
-    def test_successful_pause(self, scan_job, source):
-        """Create a scan task and status_pause it."""
-        task = ScanTask.objects.create(
-            job=scan_job,
-            source=source,
-            scan_type=ScanTask.SCAN_TYPE_CONNECT,
-            status=ScanTask.PENDING,
-        )
-        task.status_pause()
-        assert messages.ST_STATUS_MSG_PAUSED == task.status_message
-        assert task.status == ScanTask.PAUSED
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "model_action,expected_status,expected_message,check_start_time,check_end_time",
+    [
+        [
+            ScanTask.status_start,
+            ScanTask.RUNNING,
+            messages.ST_STATUS_MSG_RUNNING,
+            True,
+            False,
+        ],
+        [
+            ScanTask.status_restart,
+            ScanTask.PENDING,
+            messages.ST_STATUS_MSG_RESTARTED,
+            False,
+            False,
+        ],
+        [
+            ScanTask.status_pause,
+            ScanTask.PAUSED,
```

Comment:
mmmm we still need some PAUSE/RESTART cleanup.

(not that this needs to be done in this PR. just thinking out loud...)

Reply:
I thought about that when I saw this, too, but I didn't want to scope creep here today. The old tests tested pause/restart. So, the new tests do too.